vm/src/stdlib/os.rs

```diff
@@ -1127,6 +1127,11 @@ pub(super) mod _os {
         OutputMode::String.process_path(curdir_inner(vm)?, vm)
     }
 
+    #[pyfunction]
+    fn fork(vm: &VirtualMachine) -> PyObjectRef {
+        unsafe { vm.ctx.new_int(libc::fork()).into() }
```

Comment:
So, slightly looking into this it seems that you'll need to do a couple things:

1. Add the `register_at_fork` function. This does seem to be slightly straight-forward, logic wise at lease. You take in three callables and, after some sanity checks, store them in the interpreter state (typically `VirtualMachine`'s [`PyGlobalState`](https://github.com/RustPython/RustPython/blob/d918f7e51bf5d229b144bec686b98678f9f363e5/vm/src/vm/mod.rs#L89) in our case). This can ideally be a struct but using a vec a-la `atexit_funcs` could be another approach if the `struct` is stopping you.
2. Modify `fork` to use these. Looking at [CPython's implementation](https://github.com/python/cpython/blob/a210cac77649a18bf0eb7354c9187ec91ccb078a/Modules/posixmodule.c#L7480) this again seems relatively straight-forward. Call `before` before you issue the `fork` call and based on if you're the child/parent (`pid == 0`) issue either `after_in_child`/`after_in_parent`.

Reply:
Thanks, Point 2 looks easy but I was stuck point 1, but now I can move ahead with this pointer.